services/QuillLMS/engines/evidence/app/services/evidence/gen_ai/secondary_feedback_prompt_builder.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Evidence
+  module GenAI
+    class SecondaryFeedbackPromptBuilder < PromptBuilder
+      TEMPLATE_FOLDER = 'app/services/evidence/gen_ai/secondary_feedback_prompts/'
+      DEFAULT_TEMPLATE = '2024_07_17_with_highlight.md'
+      EXAMPLE_LIMIT = 100
+
+      private def template_variables
+        {
+          highlight_texts:,
+          primary_secondary_examples:,
+          plagiarism_text:
+        }
+      end
+
+      private def default_template = DEFAULT_TEMPLATE
+      private def template_folder = TEMPLATE_FOLDER
+
+      private def primary_secondary_examples
+        markdown_table_rows(feedback_data_tuples)
+      end
+
+      private def feedback_data_tuples
+        feedback_data
+          .select {|fd| fd.conjunction == prompt.conjunction }
+          .first(EXAMPLE_LIMIT)
+          .map {|f| [f.primary, f.secondary]}
+      end
+
+      private def feedback_data
+        @feedback_data ||= Evidence::GenAI::SecondaryFeedbackDataFetcher.run
```

Comment:
This seems odd not having any arguments.  I see a default FILE_TRAIN in the file, but what about other args?

Reply:
This service was originally just pulling the entire training file (or whatever file) and then the user would filter what they want. Looking at this with fresh eyes and based on this comment, I've refactored it so the `DataFetcher` does the fetching AND filtering, which I think makes more sense (and what you are driving at). Thanks for the feedback!